model/error.go

```diff
@@ -14,20 +14,49 @@
 
 package model
 
-import "fmt"
+import (
+	"encoding/json"
+	"errors"
+)
 
 type ConflictError struct {
-	Conflicts string
-	Err       error
+	Err       error       `json:"error"`
+	Metadata  interface{} `json:"metadata,omitempty"`
+	RequestID string      `json:"request_id,omitempty"`
```

Comment:
would it be possible to rely on [‎rest.utils/error.go](https://github.com/mendersoftware/go-lib-micro/blob/master/rest.utils/error.go#L17) or [rest_utils/api_error.go](https://github.com/mendersoftware/go-lib-micro/blob/master/rest_utils/api_error.go#L22) in these situations to be able to rely on a common error format? (maybe we could eventually even reference a common error definition in the api specs)

Reply:
The only issue is that the `rest[._]utils` errors does not wrap errors so the types would be in conflict. I guess I could make a new `ErrorV2` with proper initialization funcs. However, since this task originally was only 1SP, I'd like to defer that refactoring to some other time.